rpcserver.go

```diff
@@ -1306,15 +1306,76 @@ func assertAccountReady(acct *account.Account) error {
 // validateOrder validates the order to ensure that all fields are consistent,
 // and the order is likely to be accepted by the auctioneer. If this method
 // returns nil, then the order is safe to submit to the auctioneer.
-func (s *rpcServer) validateOrder(order order.Order, acct *account.Account,
+func (s *rpcServer) validateOrder(o order.Order, acct *account.Account,
 	auctionTerms *terms.AuctioneerTerms) error {
 
+	batchVersion, err := s.server.determineBatchVersion()
+	if err != nil {
+		return err
+	}
+
+	var usesAnchors bool
+	switch o.Details().ChannelType {
+	case order.ChannelTypeScriptEnforced:
+		usesAnchors = true
+	default:
+		usesAnchors = false
+	}
+
+	switch castOrder := o.(type) {
+	case *order.Ask:
+		if castOrder.AnnouncementConstraints == order.OnlyUnannounced &&
+			!batchVersion.SupportsUnannouncedChannels() {
+
+			return fmt.Errorf("batch version %v does not support "+
+				"unannounced channels", batchVersion)
+		}
+
+		if castOrder.ConfirmationConstraints == order.OnlyZeroConf {
+			if !batchVersion.SupportsZeroConfChannels() {
+				return fmt.Errorf("batch version %v does not "+
+					"support zero conf channels",
+					batchVersion)
+			}
+
+			if !usesAnchors {
+				return fmt.Errorf("zero conf channels need " +
+					"to use anchors. Use the script " +
+					"enforced channel type")
+			}
+		}
+
+	case *order.Bid:
+		if castOrder.UnannouncedChannel &&
+			!batchVersion.SupportsUnannouncedChannels() {
```

Comment:
I don't think this specific case can happen, since support for unannounced channels is purely a question of using the latest Pool version (which this check also is included in).

Reply:
Same with the unannounced check in the ask order.